build-logic/conventions/src/main/kotlin/com/squareup/anvil/conventions/RootPlugin.kt

```diff
@@ -0,0 +1,62 @@
+package com.squareup.anvil.conventions
+
+import com.rickbusarow.kgx.checkProjectIsRoot
+import com.squareup.anvil.benchmark.BenchmarkPlugin
+import com.squareup.anvil.conventions.utils.isInDelegateBuild
+import com.squareup.anvil.conventions.utils.isInMainAnvilBuild
+import com.squareup.anvil.conventions.utils.libs
+import org.gradle.api.Project
+
+open class RootPlugin : BasePlugin() {
+  override fun Project.jvmTargetInt(): Int = libs.versions.jvm.target.minimal.get().toInt()
+  override fun beforeApply(target: Project) {
+
+    target.checkProjectIsRoot { "RootPlugin must only be applied to the root project" }
+
+    target.plugins.apply(BenchmarkPlugin::class.java)
+    target.plugins.apply("java-base")
+
+    if (target.gradle.includedBuilds.isNotEmpty()) {
+      target.plugins.apply(CompositePlugin::class.java)
+    }
+  }
+
+  override fun afterApply(target: Project) {
+
+    target.logVersionInfo()
+
+    if (target.isInMainAnvilBuild() || target.isInDelegateBuild()) {
```

Comment:
What are the other options beyond main build and delegate build here?

Reply:
Once upon a time...

I forgot to clean this up. :)